editor/src/core/shared/array-utils.ts

```diff
@@ -285,3 +305,45 @@ export function safeIndex<T>(array: Array<T>, index: number): T | undefined {
     return undefined
   }
 }
+
+export function intersection<T>(
+  first: Array<T>,
+  second: Array<T>,
+  eqFn: (l: T, r: T) => boolean = is,
+): Array<T> {
+  let result: Array<T> = []
+  for (const valueFromFirst of first) {
+    let foundIntersection: boolean = false
+    for (const valueFromSecond of second) {
+      foundIntersection = eqFn(valueFromFirst, valueFromSecond)
+      if (foundIntersection) {
+        let shouldAdd: boolean = true
+        for (const valueFromResult of result) {
+          if (eqFn(valueFromResult, valueFromFirst)) {
+            shouldAdd = false
+            break
+          }
+        }
+        if (shouldAdd) {
+          result.push(valueFromFirst)
+        }
+      }
+    }
+    if (foundIntersection) {
+      continue
```

Comment:
If I read this correctly, this continue will do nothing as this is the last statement of the `for` loop

Reply:
Hah, that's hilarious, I must've had something after this originally.

Reply:
Fixed in 82e3a90.